iravelmakina/linear-transformations-ivelmakina
# Part two

### Description:
- Added OpenCV-based functions to perform rotation, scaling, reflection, and shearing of points and images.

### Changes:
- Implemented functions to rotate, scale, reflect, and shear points using OpenCV (`rotate_points_cv`, `scale_points_cv`, `reflect_points_cv`, `shear_points_cv`).
- Implemented a function to transform points using a custom matrix with OpenCV (`transform_points_cv`).
- Added corresponding examples of point transformations using the OpenCV functions.
- Implemented functions to rotate, scale, reflect, and shear images using OpenCV (`rotate_image`, `scale_image`, `reflect_image`, `shear_image`).
- Added corresponding examples of image transformations using the OpenCV functions.
- Changed matrix output to the console, rounding all the coordinates and ensuring user-friendly appearance.

### How to Test:
1. Ensure that OpenCV is properly installed in your Python environment.
2. Run the provided examples of point and image transformations using the OpenCV functions.
3. Verify that the transformations are applied correctly to the points and images.
4. Compare the results with the expected transformations.
5. Adjust the parameters of the transformations (e.g., angle, scale factor, shear factor) to observe different effects.
6. Check that the provided examples cover various scenarios of point and image transformations.

### Screenshots:
1. Matrix transformations (compared with results from part one below):
1.1
![Untitled (5)](https://github.com/iravelmakina/linear-transformations-ivelmakina/assets/150923968/a6293be4-b685-4c49-8c88-3c4ef1f63f6b)
1.2
![Untitled](https://github.com/iravelmakina/linear-transformations-ivelmakina/assets/150923968/b6113be2-41ae-42a5-93ee-a729dbf2099a)
1.3
![Untitled (1)](https://github.com/iravelmakina/linear-transformations-ivelmakina/assets/150923968/3a4f5c93-ad62-4f10-b01e-44ff96bd09d6)
1.4
![Untitled (2)](https://github.com/iravelmakina/linear-transformations-ivelmakina/assets/150923968/54ef54b9-660c-4daa-9011-a3f50b9b4e6d)
1.5
![Untitled (4)](https://github.com/iravelmakina/linear-transformations-ivelmakina/assets/150923968/aa5c90cb-5f15-4010-9d92-ff076761474a)


3. Image transformations:
![Screenshot 2024-06-01 152513](https://github.com/iravelmakina/linear-transformations-ivelmakina/assets/150923968/c4fb8b80-d7ee-4574-a13e-96c2f45a4b32)
![Screenshot 2024-06-01 152542](https://github.com/iravelmakina/linear-transformations-ivelmakina/assets/150923968/600baac0-b5fe-4a77-b9f0-2d0527059838)
![Screenshot 2024-06-01 152555](https://github.com/iravelmakina/linear-transformations-ivelmakina/assets/150923968/075fa468-47f4-4e0a-a22f-ce8694a5b5b5)
![Screenshot 2024-06-01 152604](https://github.com/iravelmakina/linear-transformations-ivelmakina/assets/150923968/0bc685bd-7cde-469b-8a0b-c4729b393b3f)
![Screenshot 2024-06-01 152611](https://github.com/iravelmakina/linear-transformations-ivelmakina/assets/150923968/255af56b-8558-42b3-9bdb-2e97a76d0505)
![Screenshot 2024-06-01 152619](https://github.com/iravelmakina/linear-transformations-ivelmakina/assets/150923968/317d4348-c21b-4305-8771-597dfa507dfa)
![Screenshot 2024-06-01 152625](https://github.com/iravelmakina/linear-transformations-ivelmakina/assets/150923968/327167b4-85c4-4a19-ae5b-bfa39b63fff7)
